search-enterprise/latest/config/transactional.md

```diff
@@ -0,0 +1,307 @@
+
+TMDQ supports use cases where eventual consistency is not the preferred option.
+
+The Elasticsearch subsystem is eventually consistent. The amount of time a change a change takes to reflect in the index is normally less than 1 second, but can be longer under heavy load, or for complex/cascading updates. Elasticsearch indexes the metadata and the content of each updated node, in the order in which the nodes were last changed. The indexing components will try to index information about nodes as fast as possible, but content indexing is likely to be limited by the time needed to extract text from the files and all indexing will be affected by the rate at which the nodes are being changed.
+
+Some queries can be executed both transactionally against the database or with eventual consistency against the Elasticsearch index. Only a subset of queries using the AFTS or CMIS query languages can be executed against the database. No queries using the Lucene query language can be used against the database whereas, `selectNodes` (XPATH) on the Java API always goes against the database, walking and fetching nodes as required.
+
+In general, TMDQ does not support:
+
+* Structural queries.
+* Full text search.
+* Special fields (For example `SITE`).
+* Faceting.
+* Any aggregation.
+   > **Note:** This includes counting the total number of matches for the query.
+
+AFTS and CMIS queries are parsed to an abstract form. This is then sent to an execution engine. There are two execution engines: the database and the Elasticsearch index. The default is to try the database first and fall back to the Elasticsearch index, if the query is not supported against the database. This is configurable for a search subsystem and per query using the Java API.
+
+> **Note:** Alfresco Content Services supports TMDQ by default.
+
+## Options supported by Query Languages
+
+Use this information to know what options are supported by the v1 REST API, CMIS Query Language (QL), and Alfresco Full Text Search Query Language (FTS QL).
+
+### v1 REST API and TMDQ
+
+For the v1 REST API, anything that is not a simple query, a filter query, an option that affects these, or an option that affects what is returned for each node in the results, is not supported by TMDQ.
+
+TMDQ supports:
+
+* `query`
+* `paging`
+* `include`
+* `includeRequest`
+* `fields`
+* `sort`
+* `defaults`
+* `filterQueries`
+* `scope` (single)
+* `limits` for permission evaluation
+
+The default limits for permission evaluation will restrict the results returned from TMDQ based on both the number of results processed and time taken. These can be increased, if required.
+
+The v1 REST API does not support TMDQ for:
+
+* `templates`
+* `localisation` and `timezone`
+* `facetQueries`
+* `facetFields`
+* `facetIntervals`
+* `pivots`
+* `stats`
+* `spellcheck`
```

Comment:
I don't think this is currently supported by the Elasticsearch Connector at all, and I'm not sure about some of the other items in this list. While they are definitely all unsupported for TMDQ then they may also be currently unsupported by ES too, making the paragraph a little misleading.

Reply:
Removed.